client/client.cpp

```diff
@@ -2,105 +2,139 @@
 #include <fstream>
 #include <iostream>
 #include <numeric>
-#include <sstream>
 #include <sodium.h>
+#include <sstream>
 #include <thrift/protocol/TBinaryProtocol.h>
 #include <thrift/transport/TSocket.h>
 #include <thrift/transport/TTransportUtils.h>
 
-#include "client_utils.h"
 #include "../constants/constants.h"
 #include "../crypto/encryption_engine.h"
 #include "../gen-cpp/RPC.h"
+#include "../host/redis.h"
+#include "client_utils.h"
 
 using namespace std::chrono;
 using namespace apache::thrift;
 using namespace apache::thrift::protocol;
 using namespace apache::thrift::transport;
 
 class ClientHandler {
- private:
-	std::ifstream seed_data; 
-	std::vector<float> latencies;
-
- public:
-	ClientHandler() = default;
-
-	ClientHandler(std::string path) {
-		seed_data.open(path);
-		if (!seed_data.is_open()) {
-    	throw std::invalid_argument("Invalid path to seed data");
-		}
-	}
-
-	void run_threaded() {
-		std::vector<std::thread> threads;
-		for (int i = 0; i < NUM_CLIENTS; i++) {
-			threads.push_back(std::thread(&ClientHandler::run, this));
-		}
-
-		// Wait for all threads to finish
-		for (std::thread& thread : threads) thread.join();
-	}
-
-	void run() {
-		auto socket = std::make_shared<TSocket>(HOST_IP, HOST_PORT);
-		auto transport = std::make_shared<TBufferedTransport>(socket);
-		auto protocol = std::make_shared<TBinaryProtocol>(transport);
-		RPCClient client(protocol);
-
-		transport->open();
-
-		std::string val;
-		// If seed data exists, run the client with data
-		if (seed_data.is_open()) {
-			std::string line;
-			while (readFile(seed_data, line)) {
-				Operation op = getSeedOperation(line);
-				auto start = high_resolution_clock::now();
-				client.access(val, op);
-				auto end = high_resolution_clock::now();
-				latencies.push_back(duration_cast<microseconds>(end - start).count());
-			}
-		} 
-		// If seed data does not exist, run client on random values
-		else {
-			for (int i = 0; i < 1000; ++i) {
-				Operation op = genRandOperation();
-				auto start = high_resolution_clock::now();
-				client.access(val, op);
-				auto end = high_resolution_clock::now();
-				latencies.push_back(duration_cast<microseconds>(end - start).count());
-			}
-		}
-
-		transport->close();
-	}
-
-	void getAveLatency() {
-		std::cout << "[Client]: Data access complete, average latency: " << std::accumulate(latencies.begin(), latencies.end(), 0.0) / latencies.size() << " microseconds" << std::endl;
-	}
+  private:
+    std::ifstream seed_data;
+    bool init_db = false;
+    int num_clients = 16;
+    float p_get = 0.5;
+
+    std::vector<float> latencies;
+
+  public:
+    ClientHandler(int argc, char *argv[]) {
+        parseArgs(argc, argv, seed_data, init_db, num_clients, p_get);
+    }
+
+    void start() {
+        if (init_db) {
+            initDB();
+        } else {
+            runThreaded();
+        }
+    }
```

Comment:
Are `initDB` and `runThreaded` exclusive or can both be done in the same run? We don't necessarily have to change this right now, but as a future improvement PR would be good

Reply:
Both `initDB` and `runThreaded` can be run without any seed data. If we want them be done in same run we would have implement additional flags for `data-access` similar to `--initdb` and also a `operations_file` flag to get path to operations. 

I think it might just be easier to keep them exclusive given init db will be used less often

Reply:
Sounds good, we can revisit this if time permits